### PR TITLE
Fixed some common grammatical errors in the comments.

### PR DESCRIPTION
**Pull request by OneUpWallStreet**
Fixed some common grammatical mistakes found in the comments of some files.

### CODE REVIEW

**Comment by OneUpWallStreet** [comment]
@swift-ci please smoke test

**Comment by OneUpWallStreet** [comment]
@swift-ci smoke test

**Comment by OneUpWallStreet** [comment]
@swift-ci Please smoke test

**Comment by OneUpWallStreet** [comment]
@swift-ci Please smoke test

File: lib/SILOptimizer/Transforms/SimplifyCFG.cpp [hunk 1]
```diff
@@ -663,7 +663,7 @@ bool SimplifyCFG::dominatorBasedSimplify(DominanceAnalysis *DA) {
 
   // Split all critical edges such that we can move code onto edges. This is
   // also required for SSA construction in dominatorBasedSimplifications' jump
-  // threading. It only splits new critical edges it creates by jump threading.
+  // threading. It only splits new critical edges and it creates by jump threading.
```
Comment (xedin) [inline lib/SILOptimizer/Transforms/SimplifyCFG.cpp:666]: This new ‘and’ changes meaning of the sentence.

File: lib/SILOptimizer/Transforms/SimplifyCFG.cpp [hunk 2]
```diff
@@ -2053,8 +2053,8 @@ static bool hasSameUltimateSuccessor(SILBasicBlock *noneBB, SILBasicBlock *someB
     return true;
 
   // Otherwise, lets begin a traversal along the successors of noneSuccessorBB,
-  // searching for someSuccessorBB, being careful to only allow for blocks to be
-  // visited once. This enables us to guarantee that there are not any loops or
+  // searching for someSuccessorBB, being careful such that we only allow for blocks to be
```
Comment (xedin) [inline lib/SILOptimizer/Transforms/SimplifyCFG.cpp:2056]: Why such that is better than to in this case?

File: lib/SILOptimizer/Transforms/SimplifyCFG.cpp [hunk 3]
```diff
@@ -4048,14 +4048,14 @@ bool SimplifyCFG::simplifyProgramTerminationBlock(SILBasicBlock *BB) {
...
+  // after this pass, that's why we do not really get the benefit of reusing the
   // computation for the next iteration of the pass.
   if (!isARCInertTrapBB(BB))
     return false;
 
   // This is going to be the last basic block this program is going to execute
-  // and this block is inert from the ARC's prospective, no point to do any
-  // releases at this point.
+  // and this block is inert from the ARC's prospective,so there's no point to do any
+  // releases beyond/at this point.
```
Comment (xedin) [inline lib/SILOptimizer/Transforms/SimplifyCFG.cpp:4058]: Why beyond is necessary here?

File: include/swift/Sema/ConstraintSystem.h [hunk 4]
```diff
@@ -2521,8 +2521,8 @@ class ConstraintSystem {
 
   private:
     /// The list of constraints that have been retired along the
-    /// current path, this list is used in LIFO fashion when constraints
-    /// are added back to the circulation.
+    /// current path, this list is used in LIFO (Last In First Out)
```
Comment (xedin) [inline include/swift/Sema/ConstraintSystem.h:2524]: I don't think expansion of abbreviation adds much here.

File: include/swift/Sema/ConstraintSystem.h [hunk 5]
```diff
@@ -2821,8 +2821,8 @@ class ConstraintSystem {
   /// able to emit an error message, or false if none of the fixits worked out.
   bool applySolutionFixes(const Solution &solution);
 
-  /// If there is more than one viable solution,
-  /// attempt to pick the best solution and remove all of the rest.
+  /// If there is more than one viable solution, we attempt
```
Comment (xedin) [inline include/swift/Sema/ConstraintSystem.h:2824]: Let’s not use ‘we’ here

**Review by xedin — COMMENTED** [review]
Mostly looks good, thank you! I left some comments inline.

File: lib/SILOptimizer/Transforms/SimplifyCFG.cpp (hunk 1 above)
Comment (OneUpWallStreet) [reply lib/SILOptimizer/Transforms/SimplifyCFG.cpp]: > Mostly looks good, thank you! I left some comments inline.

Hey @xedin I'm sorry about those errors I'm a little new to this, but I think have made all the changes you mentioned! Please let me know if there's anything else I need to do 😄

**Review by xedin — COMMENTED** [review]
Looks much better now,  thank you! There is only a couple of newline changes I don’t think are necessary.

**Comment by OneUpWallStreet** [comment]
> Looks much better now, thank you! There is only a couple of newline changes I don’t think are necessary.

Thank you @xedin, I removed the extra newlines. Please let me know if there's anything else I need to do.

**Review by xedin — APPROVED** [review]

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by OneUpWallStreet** [comment]
> @swift-ci please smoke test

Thank you for taking the time and approving my request @xedin 😄

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test

File: lib/SILOptimizer/Transforms/SimplifyCFG.cpp [hunk 6]
```diff
@@ -2054,7 +2054,7 @@ static bool hasSameUltimateSuccessor(SILBasicBlock *noneBB, SILBasicBlock *someB
 
   // Otherwise, lets begin a traversal along the successors of noneSuccessorBB,
   // searching for someSuccessorBB, being careful to only allow for blocks to be
-  // visited once. This enables us to guarantee that there are not any loops or
+  // visited once. This enables us to guarantee that there no loops or
```
Comment (xwu) [inline lib/SILOptimizer/Transforms/SimplifyCFG.cpp:2057]: This is missing a word.

**Pull request merged**